numpy/core/src/multiarray/buffer.c

```diff
@@ -215,19 +215,19 @@ _buffer_format_string(PyArray_Descr *descr, _tmp_string_t *str,
             subarray_tuple = Py_BuildValue("(O)", descr->subarray->shape);
         }
 
-        _append_char(str, '(');
+        if (_append_char(str, '(') < 0) return -1;
         for (k = 0; k < PyTuple_GET_SIZE(subarray_tuple); ++k) {
             if (k > 0) {
-                _append_char(str, ',');
+                if (_append_char(str, ',') < 0) return -1;
             }
             item = PyTuple_GET_ITEM(subarray_tuple, k);
             dim_size = PyNumber_AsSsize_t(item, NULL);
 
             PyOS_snprintf(buf, sizeof(buf), "%ld", (long)dim_size);
-            _append_str(str, buf);
+            if (_append_str(str, buf) < 0) return -1;
             total_count *= dim_size;
         }
-        _append_char(str, ')');
+        if (_append_char(str, ')') < 0) return -1;
 
         Py_DECREF(subarray_tuple);
```

Comment:
Memory leak here if the new error path is taken.

Reply:
And we never put the statement on the same line :)
```
        if (_append_char(str, ')') < 0) {
            return -1;
        }
```